src/chart/generateCategoricalChart.tsx

```diff
@@ -1415,6 +1420,7 @@ export const generateCategoricalChart = ({
 
     clearDeferId = () => {
       if (!isNil(this.deferId) && deferClear) {
+        // @ts-expect-error types miss the relationship between deferClear and defer
```

Comment:
What do you suggest we do with these statements? 
Should we create a follow up issue? 

I am a bit sceptical of the value of introducing types if that requires us to also `@ts-expect-error`.

Reply:
Do you see a way how you could still fix these in this PR?

Reply:
These two are forming a pair and if we want the pair to resolve to the same generic then they need to be in the same generic container - like an object or a Class. I can fix this one.